### ISSUE TITLE
Editing a table with a row type of file, saves the value as "[]"

### ISSUE THREAD

**Issue by danord24** [thedevdojo/voyager#3097]
- Laravel Version: 5.6
- Voyager Version: 1.1.0
- PHP Version: 7.1
- Database Driver & Version: MySQL 5.7.19

### Description:
When editing a table with a row->type of "file". If you already have a file uploaded and have changed other content but have left the file as-is, the file field is updated to a string of "[]".

### Steps To Reproduce:

1. Using the BREAD builder, create a new entry in a table which has a row type of File. Populate all fields, then save.
2. Edit the new entry, change some fields, but leave the file field as-is. It should still reference the original file you uploaded, then save.
3. The file field should now have been saved as "[]", meaning you have lost the refence to the original file and its path etc.

I have found a fix for this, but I'm not sure if this needs looking into further down the line.

In vendor/tgc/voyager/src/Http/Controllers/Controller.php - insertUpdateData(), Line 88.

Change:

`if (is_null($content)) {`

to 

`if (is_null($content) || $content == '[]') {`

This check catches the empty json array and passes in the orignal value as intended.

```
// If the file upload is null and it has a current file keep the current file
if ($row->type == 'file') {
  $content = $data->{$row->field};
}
```

**Comment by b4dnewz** [comment]
I confirm the issue, when a **file** type row is saved on the update page it lose the original value.

**Comment by butaminas** [comment]
I think this is a bigger issue then it looks like and it is weird that it haven't been fixed yet as it makes Settings page file type unusable. The OP fix:

> In vendor/tgc/voyager/src/Http/Controllers/Controller.php - insertUpdateData(), Line 88.

didn't worked for me. In fact, this Controller doesn't seem to have any impact on saving changes in Settings page (perhaps because my problem is in settings page and not BREAD editor?).

**What did worked for me was this:**
In `vendor/tcg/voyager/src/Http/Controllers/VoyagerSettingsController.php`
where
`if ($setting->type == 'image' && $content == null)`
changed it to
`if (($setting->type == 'image' && $content == null) || ($setting->type == 'file' && $content == '[]'))`


Also, I noticed that each setting from all tabs are being saved when you save settings. Shouldn't it save settings from the tab that is active or even better, only saving the settings that are actually changed?

**Comment by emptynick** [comment]
@butaminas thats because Settings is different from "normal" BREAD. With that its out of scope of this issue and has actually been [fixed already](https://github.com/the-control-group/voyager/pull/3060)

**Comment by emptynick** [comment]
Fixed in #3521
